### PR TITLE
defining cascade classifier accuracy

### PR DESCRIPTION
**Pull request by StevenPuttemans**
Allowing people to manually define how sharp a cascade classifier model should be trained
2 years of experimenting with the cascade classifiers led me to believe that trying to built beyond a reported acceptanceRatio of 10e-5 after a set of weak classifier stages, only results in a classifier which is dramatically overfitting on the training data.

This patch allows users of the algorithm to manually define a value where they want training to back off. If nothing supplied than -1 is used and the influence of this value is ignored!

### CODE REVIEW

**Comment by StevenPuttemans** [comment]
@vpisarev are you in agreement with this? Should I provide something extra?

File: apps/traincascade/cascadeclassifier.cpp [hunk 1]
```diff
@@ -207,13 +209,18 @@ bool CvCascadeClassifier::train( const string _cascadeDirName,
...
             break;
         }
         if( tempLeafFARate <= requiredLeafFARate )
         {
             cout << "Required leaf false alarm rate achieved. "
-                 "Branch training terminated." << endl;
+                    "Branch training terminated." << endl;
+            break;
+        }
+        if( (tempLeafFARate <= acceptanceRatioBreakValue) && (acceptanceRatioBreakValue != -1.0) ){
```
Comment (vpisarev) [inline apps/traincascade/cascadeclassifier.cpp]: a minior request. let's use "acceptanceRatioBreakValue &lt; 0" instead of "acceptanceRatioBreakValue != -1.0"

**Comment by vpisarev** [comment]
@StevenPuttemans, sorry for delay; I'm basically fine with the patch, with one minor suggestion (see above).

**Comment by StevenPuttemans** [comment]
PR fixed and rebased! I will supply the same extra for the 2.4 branch!

**Comment by StevenPuttemans** [comment]
@vpisarev please wait before merging both PRs, I will first add documentation for both branches, forgot that one!

**Comment by StevenPuttemans** [comment]
Done!

**Comment by vpisarev** [comment]
:+1:

**Comment by vpisarev** [comment]
oops, looks like the condition should be inversed: "&lt; 0" to "&gt;= 0"

**Comment by StevenPuttemans** [comment]
I will supply a fix! Indeed this is wrong :D
